src/common/btree.c

```diff
@@ -847,10 +919,18 @@ btr_root_init(struct btr_context *tcx, struct btr_root *root, bool in_place)
 	root->tr_class		= tcx->tc_class;
 	root->tr_feats		= tcx->tc_feats;
 	root->tr_order		= tcx->tc_order;
-	if (tcx->tc_feats & BTR_FEAT_DYNAMIC_ROOT)
-		root->tr_node_size	= 1;
-	else
+	if (tcx->tc_feats & BTR_FEAT_DYNAMIC_ROOT) {
+		/** If the first entry will be embedded, we'll need to insert 2
+		 *  entries, so set the initial size accordingly.  At present,
+		 *  we always go from 1 to 3.
+		 */
+		if (tcx->tc_feats & BTR_FEAT_EMBED_FIRST)
+			root->tr_node_size = MIN(3, tcx->tc_order);
```

Comment:
BTR_ORDER_MIN is 3 so it can't be less than 3.
This is just setting what the first dynamic allocation will be.

I decided not to handle the downgrade case with the assumption that if we allocated full tree once, we will do it again.  Maybe a bad assumption but not without some reasoning.  I think the common case will be a single entry.